### PR TITLE
docs: update reduction docs

### PR DESCRIPTION
**Pull request by prmukherj**
Update the reduction docs to highlight the functional usage (particularly in cases with multiple solvers).

Correct and update the field_data docs as well.

Update reduction tests to highlight the functional behaviour.

### CODE REVIEW

File: doc/source/user_guide/fields/reduction.rst [hunk 1]
```diff
@@ -3,350 +3,302 @@
...
-  >>> case_path = download_file("Static_Mixer_main.cas.h5", "pyfluent/static_mixer")
-  >>> solver.settings.file.read(file_type="case", file_name=case_path)
+    >>> import ansys.fluent.core as pyfluent
+    >>> from ansys.fluent.core.solver.function import reduction
+    >>> from ansys.fluent.core.examples import download_file
 
+    >>> solver1 = pyfluent.launch_fluent(mode=pyfluent.FluentMode.SOLVER)
+    >>> case_path = download_file("Static_Mixer_main.cas.h5", "pyfluent/static_mixer")
+    >>> solver1.settings.file.read(file_type="case", file_name=case_path)
+    >>> solver1.settings.solution.initialization.hybrid_initialize()
```
Comment (mkundu1) [inline doc/source/user_guide/fields/reduction.rst:29]: In general, for any fields API we should demostrate the usage with converged data. The initialized data will have uniform distribution of pressure, so it doesn't make much sense to take average/min etc.

File: doc/source/user_guide/fields/reduction.rst [hunk 2]
```diff
@@ -3,350 +3,302 @@
...
-  >>> solver = pyfluent.launch_fluent(mode=pyfluent.FluentMode.SOLVER)
-  >>> case_path = download_file("Static_Mixer_main.cas.h5", "pyfluent/static_mixer")
-  >>> solver.settings.file.read(file_type="case", file_name=case_path)
+    >>> import ansys.fluent.core as pyfluent
+    >>> from ansys.fluent.core.solver.function import reduction
+    >>> from ansys.fluent.core.examples import download_file
 
+    >>> solver1 = pyfluent.launch_fluent(mode=pyfluent.FluentMode.SOLVER)
+    >>> case_path = download_file("Static_Mixer_main.cas.h5", "pyfluent/static_mixer")
+    >>> solver1.settings.file.read(file_type="case", file_name=case_path)
```
Comment (mkundu1) [inline doc/source/user_guide/fields/reduction.rst:28]: Please use concise form of the API whereever possible:
```
solver1 = pyfluent.launch_fluent()
solver1.settings.file.read_case(file_name=...)
```

File: doc/source/user_guide/fields/reduction.rst [hunk 3]
```diff
@@ -3,350 +3,302 @@
...
 .. code-block:: python
 
-  >>> solver.fields.reduction.moment(
-  >>>   expression="['inlet1']",
-  >>>   locations=[solver.settings.setup.boundary_conditions.velocity_inlet["inlet2"]]
+  >>> cent = reduction.centroid(
+  >>>   locations=[VelocityInlets(settings_source=solver) for solver in [solver1, solver2]]
   >>> )
-  [ 1.15005117e-24,  1.15218653e-24, -6.60723735e-20]
+  >>> cent.array
```
Comment (mkundu1) [inline doc/source/user_guide/fields/reduction.rst:286]: I'd prefer a simple Python list over a structure behind a private class.

The docstring of centroid says it returns a float.

File: doc/source/user_guide/index.rst [hunk 4]
```diff
@@ -47,7 +47,7 @@ A simple example
   >>> setup.cell_zone_conditions.set_zone_type(zone_list="elbow-fluid", new_type="fluid")
   >>> solution.initialization.hybrid_initialize()
   >>> solution.run_calculation.iterate(iter_count=100)
-  >>> velocity_data = solver.fields.field_data.get_vector_field_data(field_name="velocity", surface_name="cold-inlet")
+  >>> velocity_data = solver.fields.field_data.get_vector_field_data(field_name="velocity", surfaces=["cold-inlet"])
```
Comment (mkundu1) [inline doc/source/user_guide/index.rst:50]: There are some more in file session user-guide

Comment (prmukherj) [reply doc/source/user_guide/index.rst:50]: ok, will look into it.

File: doc/source/user_guide/fields/reduction.rst (hunk 1 above)
Comment (prmukherj) [reply doc/source/user_guide/fields/reduction.rst:29]: Thanks @mkundu1, updated the docs to use converged data.

File: doc/source/user_guide/fields/reduction.rst (hunk 2 above)
Comment (prmukherj) [reply doc/source/user_guide/fields/reduction.rst:28]: Sure, done, Thank you.

File: doc/source/user_guide/index.rst (hunk 4 above)
Comment (prmukherj) [reply doc/source/user_guide/index.rst:50]: Done, Thank you @mkundu1.

File: doc/source/user_guide/fields/reduction.rst (hunk 3 above)
Comment (prmukherj) [reply doc/source/user_guide/fields/reduction.rst:286]: Done, Thank you.

File: doc/source/user_guide/fields/reduction.rst [hunk 5]
```diff
@@ -3,357 +3,311 @@
...
------------------------------
+Introduction to Reduction Functions
+-----------------------------------
 
-In order to access reduction function, import it and launch the Fluent solver.
-Then, make boundary conditions data, etc. available (for example, by reading case files):
+Reduction functions perform operations like computing averages, integrals, and sums over specified data locations,
+such as areas or volumes.
+
+For demonstrating the following examples, let us initialize two separate solver sessions
```
Comment (seanpearsonuk) [inline doc/source/user_guide/fields/reduction.rst:17]:
```suggestion
To demonstrate the following examples, first initialize two separate solver sessions
```
(Vale warning)

**Review by seanpearsonuk — APPROVED** [review]

**Review by mkundu1 — APPROVED** [review]

**Pull request merged**